### PR TITLE
[Worker]: BugFix code upload worker for limited nodes

### PR DESCRIPTION
**Pull request by savish28**
Context: Code upload worker.

This PR solves the following bugs

- [x] Prevent Logs in code upload worker being printed twice.
- [x] Fix read_job function in case of job not present.
- [x] Prevent failure of submission when pod is in pending state. [nodes are fully utilized and a new job is created]

Detailed Explanation for point 3:
This scenario occurs when the job is created but the number of nodes available are fully utilized, so the job is created but the corresponding pod is in a pending state in which the container statuses does not exist. 

@KhalidRmb 
cc: @Ram81

### CODE REVIEW

File: scripts/workers/code_upload_submission_worker.py [hunk 1]
```diff
@@ -550,38 +552,60 @@ def update_failed_jobs_and_send_logs(
...
-                                )
-                            )
-                            pod_log = pod_log_response.data.decode("utf-8")
-                            clean_submission = True
-                            submission_error = pod_log
-                        except client.rest.ApiException as e:
-                            logger.exception(
-                                "Exception while reading Job logs {}".format(e)
-                            )
+        if job_def:
```
Comment (Ram81) [inline scripts/workers/code_upload_submission_worker.py:555]: @savish28 `read_job` will return a valid kubernetes job if the job is in pending state. Have you tested this change?

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: scripts/workers/code_upload_submission_worker.py (hunk 1 above)
Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:555]: @Ram81 this is the case when I manually delete the job from kubectl and in this case read_job would return None, this is a different usecase.

Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:555]: Yeah, I have tested this change

Comment (Ram81) [reply scripts/workers/code_upload_submission_worker.py:555]: Is this the reason why fargate worker was failing?

Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:555]: Not sure, why the fargate was restarting is still a mystery

**Review by Ram81 — APPROVED** [review]
LGTM

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**